libsolidity/formal/SMTChecker.cpp

```diff
@@ -752,7 +770,14 @@ bool SMTChecker::createVariable(VariableDeclaration const& _varDecl)
 	if (SSAVariable::isSupportedType(_varDecl.type()->category()))
 	{
 		solAssert(m_variables.count(&_varDecl) == 0, "");
-		m_variables.emplace(&_varDecl, SSAVariable(_varDecl, *m_interface));
+		solAssert(m_stateVariables.count(&_varDecl) == 0, "");
+		if (_varDecl.isLocalVariable())
+			m_variables.emplace(&_varDecl, SSAVariable(_varDecl, *m_interface));
+		else
+		{
+			solAssert(_varDecl.isStateVariable(), "");
+			m_stateVariables.emplace(&_varDecl, SSAVariable(_varDecl, *m_interface));
```

Comment:
Why do we have this distinction in the first place? If we do not add the state variables to `m_variables`, they are not taken into account when unifying branches.

Reply:
The distinction is because state vars are visited before functions, so we keep a single copy while visiting all the functions, and add them to `m_variables` in the beginning of each function

Reply:
So, the thing about state vars being visited before functions: has to be done manually